apis/v1beta1/validation/gateway.go

```diff
@@ -42,6 +44,9 @@ var (
 		gatewayv1b1.HTTPSProtocolType: {},
 		gatewayv1b1.TLSProtocolType:   {},
 	}
+
+	validHostnameAddress   = `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`
```

Comment:
Is this regular expression pattern specifically copied from a reference somewhere?

Reply:
I think we should replace it with the pattern we already have. I think we should use this pattern: https://github.com/kubernetes-sigs/gateway-api/blob/16c54cdd2546bce917c1d045a5dc6a49e51625ae/apis/v1beta1/shared_types.go#L381-L382

Reply:
sure, replaced